system/logaccess/ElasticStack/ElasticStackLogAccess.cpp

```diff
@@ -520,11 +605,75 @@ cpr::Response ElasticStackLogAccess::performESQuery(const LogAccessConditions &
 bool ElasticStackLogAccess::fetchLog(const LogAccessConditions & options, StringBuffer & returnbuf, LogAccessLogFormat format)
 {
     cpr::Response esresp = performESQuery(options);
-    processESJsonResp(esresp, returnbuf, format);
+    processESSearchJsonResp(esresp, returnbuf, format);
 
     return true;
 }
 
+class ELASTICSTACKLOGACCESS_API ElasticStackLogStream : public CInterfaceOf<IRemoteLogAccessStream>
+{
+public:
+    virtual bool readLogEntries(StringBuffer & record, unsigned & recsRead, unsigned & totalRecs) override
+    {
+        Json::Value res;
+        recsRead = 0;
+        totalRecs = 0;
+
+        if (m_esSroller.next(res))
+        {
+
+            if (!res["total"]["value"].empty())
+            {
+                totalRecs = res["total"]["value"].asUInt();
+            }
+
+            if (!res["hits"].empty())
+            {
+                recsRead = res["hits"].size();
+                std::ostringstream sout;
+                m_jsonWriter->write(res, &sout);
+                processESScrollJsonResp(sout.str().c_str(), record, m_outputFormat);
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    ElasticStackLogStream(std::string & queryString, const char * connstr, const char * indexsearchpattern, LogAccessLogFormat format,  std::size_t pageSize, std::string scrollTO)
+     : m_esSroller(std::make_shared<elasticlient::Client>(std::vector<std::string>({connstr})), pageSize, scrollTO)
+    {
+        m_outputFormat = format;
+        m_esSroller.init(indexsearchpattern, DEFAULT_ES_DOC_TYPE, queryString);
+        m_jsonWriter = m_jsonStreamBuilder.newStreamWriter();
```

Comment:
This will leak.  Need to use std::unique_ptr.  Worth running all of this code under valgrind to see if it creates any other leaks.

Reply:
will fix